python/paddle/fluid/contrib/slim/quantization/quantization_pass.py

```diff
@@ -1183,77 +1192,141 @@ def _scale_name(self, var_name):
 
 
 class AddQuantDequantPass(object):
+    supported_quantizable_op_type = [
+        "pool2d", "elementwise_add", "concat", "softmax", "argmax", "transpose",
+        "equal", "gather", "greater_equal", "greater_than", "less_equal",
+        "less_than", "mean", "not_equal", "reshape", "reshape2",
+        "bilinear_interp", "nearest_interp", "trilinear_interp", "slice",
+        "squeeze", "elementwise_sub"
+    ]
+    # list op real input and output names, to avoid processing input
+    # such as AxisTensor.
+    op_real_in_out_name = {
+        "conv2d": [["Input"], ["Output"]],
+        "depthwise_conv2d": [["Input"], ["Output"]],
+        "mul": [["X", "Y"], ["Out"]],
+        "pool2d": [["X"], ["Out"]],
+        "elementwise_add": [["X", "Y"], ["Out"]],
+        "concat": [["X"], ["Out"]],
+        "softmax": [["X"], ["Out"]],
+        "argmax": [["X"], ["Out"]],
+        "transpose": [["X"], ["Out"]],
+        "equal": [["X", "Y"], ["Out"]],
+        "gather": [["X"], ["Out"]],
+        "greater_equal": [["X", "Y"], ["Out"]],
+        "greater_than": [["X", "Y"], ["Out"]],
+        "less_equal": [["X", "Y"], ["Out"]],
+        "less_than": [["X", "Y"], ["Out"]],
+        "mean": [["X"], ["Out"]],
+        "not_equal": [["X", "Y"], ["Out"]],
+        "reshape": [["X"], ["Out"]],
+        "reshape2": [["X"], ["Out"]],
+        "bilinear_interp": [["X"], ["Out"]],
+        "nearest_interp": [["X"], ["Out"]],
+        "trilinear_interp": [["X"], ["Out"]],
+        "slice": [["Input"], ["Out"]],
+        "squeeze": [["X"], ["Out"]],
+        "elementwise_sub": [["X", "Y"], ["Out"]],
+    }
+
     def __init__(self,
                  scope=None,
                  place=None,
                  moving_rate=0.9,
                  quant_bits=8,
                  skip_pattern='skip_quant',
-                 quantizable_op_type=["elementwise_add", "pool2d"]):
+                 quantizable_op_type=["elementwise_add", "pool2d", "concat"],
+                 is_full_quantized=False):
         """
-        This pass is used to add quant_dequant op for some ops, such as the
-        'elementwise_add' and 'pool2d' op.
+        This pass add quant_dequant op for some ops, of which the inputs must be 
+        not persistable.
+        The quant_dequant op can obtain the input scale.
```

Comment:
The input scales can be obtained from the quant_dequant op.

Reply:
> The input scales can be obtained from the quant_dequant op.

Done